reactor-core/src/main/java/reactor/core/publisher/FluxIterable.java

```diff
@@ -19,16 +19,20 @@
 import java.util.Collection;
```

Comment:
I noticed `onCloseWithDropError` in both subscriptions don't call `onDiscardMultiple`. Perhaps there is a reason, but it looks like the iterator may have more items (e.g. in `poll()` if an item is `null`).

Reply:
a `QueueSubscription` must only return `null` to `poll()` if the subscription is empty/done. some cases of calling `onCloseWithDropErrors` are not suitable for attempting a drain of the iterator, for instance if `hasNext()` throws... Should we attempt to drain in the case the iterator returns a `null` value though? I'd argue this is an illegal state so the user should find the root cause of the `null` first and foremost, after which the discarding will be correctly done.